### DISCUSSION TITLE
Don't try to ssr render the `_index` page if `ssr: false` and `root.tsx` exports `HydrateFallback`

### DISCUSSION THREAD

**Discussion by remorses** [remix-run/remix#9771] [Proposals]
Currently Remix will try to prerender on the server the index route even if you have ssr set to false and export an HydrateFallback component.

Remix should use the HydrateFallback as the initial html for the index route, so the user can import client only packages in their code and don't worry about ssr.

**Comment by sergiodxa** [comment]
I agree to use the Layout + HydrateFallback if exported instead of Layout + default export, but that won't solve client-only packages breaking on import, you're just switching from pre-rendering Layout + default to pre-rendering Layout + HydrateFallback, the root module needs to be executed at build time on both cases and any import that has side effects depending on the browser will break in that case.

**Reply by remorses** [reply]
to render the root.tsx file you don't need to import other pages, so this will work most of the times
